app/controllers/bookings_controller.rb

```diff
@@ -45,16 +47,21 @@ def destroy
     @booking.destroy
     redirect_to bookings_path, notice: 'Booking was successfully destroyed.'
   end
-
+  
+  private
   def set_booking
     @booking = Booking.find(params[:id])
   end
 
-  private
+  def set_bookable
```

Comment:
Use this:

@bookable = params[:bookable_type].constantize.find(params[:bookable_id])